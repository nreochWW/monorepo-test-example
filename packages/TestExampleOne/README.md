# test-example-one

#### This is the first package I have created
