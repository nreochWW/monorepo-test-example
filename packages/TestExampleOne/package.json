{
  "name": "test-example-one",
  "description": "This is the first package I have created",
  "version": "0.0.0",
  "main": "lib/index.js",
  "license": "MIT",
  "scripts": {
    "tsc": "tsc",
    "legacy": "webpack --config legacy/webpack.config.js"
  }
}
