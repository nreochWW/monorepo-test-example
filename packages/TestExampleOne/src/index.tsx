import * as React from "react";
import styled from "styled-components";

export interface TestExampleOneProps {
  compiler: string;
  framework: string;
}

const Wrapper = styled.div`
  border: 1px solid blue;
  padding: 10px;
`;

const TestExampleOne = (props: TestExampleOneProps) => (
  <Wrapper>
    <h1>
      Hello World from {props.compiler} and {props.framework}!
    </h1>
  </Wrapper>
);

export default TestExampleOne;
