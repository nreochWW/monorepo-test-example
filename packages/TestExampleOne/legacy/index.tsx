import * as React from "react";
import * as ReactDOM from "react-dom";

import TestExampleOne from "../src/index";

ReactDOM.render(
  <TestExampleOne compiler="TypeScript" framework="React" />,
  document.getElementById("TestExampleOne")
);
