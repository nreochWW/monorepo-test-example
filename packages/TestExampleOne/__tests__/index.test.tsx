import * as React from "react";
import { shallow } from "enzyme";
import TestExampleOne from "../src/index";

describe("TestExampleOne package tests", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(
      <TestExampleOne compiler="TypeScript" framework="React" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
