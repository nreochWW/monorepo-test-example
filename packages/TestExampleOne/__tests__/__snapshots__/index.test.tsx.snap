// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestExampleOne package tests matches snapshot 1`] = `
<styled.div>
  <h1>
    Hello World from 
    TypeScript
     and 
    React
    !
  </h1>
</styled.div>
`;
